### PR TITLE
Refactor server boot code

### PR DESCRIPTION
**Pull request by jage**
* First round of refactoring, split up server setup code into `AvalancheMQ::Launcher`.
* Separate into more methods, hopefully easier to follow startup code and stack traces.
* Should make it easier to test some of this code (separate PR)

### CODE REVIEW

File: src/avalanchemq/launcher.cr [hunk 1]
```diff
@@ -0,0 +1,233 @@
+require "file"
+require "systemd"
+
+module AvalancheMQ
+  class Launcher
+    @tls_context : (OpenSSL::SSL::Context::Server|Nil)
```
Comment (carlhoerberg) [inline src/avalanchemq/launcher.cr:6]: Can add question mark instead of union of nil in crystal

File: src/avalanchemq/launcher.cr [hunk 2]
```diff
@@ -0,0 +1,233 @@
...
+    @first_shutdown_attempt = true
+    @log : Logger
+    @lock : (File|Nil)
```
Comment (carlhoerberg) [inline src/avalanchemq/launcher.cr:9]:
```suggestion
    @lock : File?
```

File: src/avalanchemq/launcher.cr (hunk 1 above)
Comment (jage) [reply src/avalanchemq/launcher.cr:6]: Ah, TIL

**Comment by jage** [comment]
Have done some tests locally (with SSL as well), seems to work. There's probably more we can do, but unless it's very weird I'd prefer to merge and do another pass later on.

**Review by carlhoerberg — APPROVED** [review]

**Pull request merged**